src/libraries/System.Private.Xml/src/System/Xml/XmlCharType.cs

```diff
@@ -39,91 +41,42 @@ internal unsafe struct XmlCharType
         // bitmap for public ID characters - 1 bit per character 0x0 - 0x80; no character > 0x80 is a PUBLIC ID char
         private const string s_PublicIdBitmap = "\u2400\u0000\uffbb\uafff\uffff\u87ff\ufffe\u07ff";
 
-        // size of XmlCharType table
-        private const uint CharPropertiesSize = (uint)char.MaxValue + 1;
 
-        // static lock for XmlCharType class
-        private static object? s_Lock;
+        public static XmlCharType Instance => default;
 
-        private static object StaticLock
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public readonly bool IsWhiteSpace(char ch)
         {
-            get
-            {
-                if (s_Lock == null)
-                {
-                    object o = new object();
-                    Interlocked.CompareExchange<object?>(ref s_Lock, o, null);
-                }
-
-                return s_Lock;
-            }
+            return (Unsafe.Add(ref MemoryMarshal.GetReference(CharProperties), ch) & fWhitespace) != 0;
```

Comment:
It would be nice to extract the unsafe code into `CharProperies(char ch)` private helper method to avoid duplicating the `Unsafe...` pattern many times.

Reply:
Done, and forced inline because it makes a significant difference if I don't.